app/views/Settings.js

```diff
@@ -52,13 +54,42 @@ export const SettingsScreen = ({ navigation }) => {
     return () => {
       BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
     };
-  }, [navigation]);
+  }, []);
+
+  //Initialyy when settings page is loaded , this method will check whether we have remove location permission from app settings.
+  // It will set status "Location Active" only if permission is granted.
+  const checkIsLocationEnable = () => {
```

Comment:
All these checks are now encoded in `LocationService.checkStatus()`: 

https://github.com/Path-Check/covid-safe-paths/blob/develop/app/services/LocationService.js#L347-L382

It was added in #749 

Please use that, it has tests. And the views do not need to access BackgroundGeolocation directly anymore.